src/Streams.Core/ParStreams.fs

```diff
@@ -132,10 +145,9 @@ module ParStream =
                         Task.Factory.StartNew(fun () ->
                                                 while partition.MoveNext() && !nextRef do
                                                     iter.Func partition.Current.Value
-                                                ())
+                                                partition.Dispose())
```

Comment:
I figured we should add a Dispose here though I don't think the IEnumerator holds any resources